api/apiclient.go

```diff
@@ -418,25 +421,27 @@ func (st *state) apiEndpoint(path, query string) (*url.URL, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	return &url.URL{
-		Scheme:   st.serverScheme,
-		Host:     st.Addr(),
-		Path:     path,
-		RawQuery: query,
-	}, nil
+	url := st.Addr()
+	url.Scheme = st.serverScheme
+	url.Path = gopath.Join(url.Path, path)
+	url.RawQuery = query
+	return url, nil
 }
 
 // ControllerAPIURL returns the URL to use to connect to the controller API.
+// The address may contain a path component.
 func ControllerAPIURL(addr string, port int) string {
+	addr, path, _ := strings.Cut(addr, "/")
```

Comment:
To be totally safe, can we validate that "addr" is not a URL string containing a scheme, eg "wss://1.1.1.1/path"